src/bin_editor/Cli.re

```diff
@@ -176,25 +155,14 @@ let parse =
   let directories = List.filter(isDirectory, absolutePaths);
   let filesToOpen = List.filter(p => !isDirectory(p), absolutePaths);
 
-  let homeOrWorkingDirectory =
-    switch (Sys.getenv_opt("HOME")) {
-    | Some(homePath) => homePath
-    | None => workingDirectory
-    };
-
-  /* Set the folder to be opened, based on 4 options:
-         - If a folder(s) is given, use the first.
-         - If no folders are given, but files are, use the dir of the first file.
-         - If no files or folders are given, and the path is, the root, "/", try and use the home directory
-         - If none of the other conditions are met, use the working directory
-     */
-
   let folder =
-    switch (directories, filesToOpen, workingDirectory) {
-    | ([hd, ..._], _, _) => hd
-    | ([], [hd, ..._], _) => Rench.Path.dirname(hd)
-    | ([], [], "/") => homeOrWorkingDirectory
-    | ([], [], workingDirectory) => workingDirectory
+    switch (directories) {
+    | [first, ..._] => Some(first)
+    | [] =>
+      switch (filesToOpen) {
+      | [first, ..._] => Some(Rench.Path.dirname(first))
+      | [] => None
+      }
```

Comment:
The CLI parsing now only deals with CLI parsing instead of making fallback decisions. If there's no valid file or folder arguments passed on the command line, `folder` will be `None` and the decision on what to do delayed so other sources (like the persistence store in this case) can be checked independently as well.

Reply:
Sounds good!